kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaLogActionUtils.java

```diff
@@ -0,0 +1,286 @@
+import io.delta.kernel.internal.util.FileNames;
+import io.delta.kernel.types.*;
+import io.delta.kernel.utils.CloseableIterator;
+import io.delta.kernel.utils.FileStatus;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
+/**
+ * Exposes APIs to read the raw actions within the *commit files* of the _delta_log. This is used
+ * for CDF, streaming, and more.
+ */
+public class DeltaLogActionUtils {
+
+  private DeltaLogActionUtils() {}
+
+  /////////////////
+  // Public APIs //
+  /////////////////
+
+  /**
+   * Represents a Delta action. This is used to request which actions to read from the commit files
+   * in {@link TableImpl#getChangesByVersion(Engine, long, long, Set)}.
+   *
+   * <p>See the Delta protocol for more details
+   * https://github.com/delta-io/delta/blob/master/PROTOCOL.md#actions
+   */
+  public enum DeltaAction {
+    REMOVE("remove", RemoveFile.FULL_SCHEMA),
+    ADD("add", AddFile.FULL_SCHEMA);
+    // Remaining actions coming in follow-up PR
+
+    public final String colName;
+    public final StructType schema;
+
+    DeltaAction(String colName, StructType schema) {
+      this.colName = colName;
+      this.schema = schema;
+    }
+  }
+
+  /**
+   * For a table get the list of commit log files for the provided version range.
+   *
+   * @param tablePath path for the given table
+   * @param startVersion start version of the range (inclusive)
+   * @param endVersion end version of the range (inclusive)
+   * @return the list of commit files in increasing order between startVersion and endVersion
+   * @throws TableNotFoundException if the table does not exist or if it is not a delta table
+   * @throws KernelException if a commit file does not exist for any of the versions in the provided
+   *     range
+   * @throws KernelException if provided an invalid version range
+   */
+  public static List<FileStatus> getCommitFilesForVersionRange(
+      Engine engine, Path tablePath, long startVersion, long endVersion) {
+
+    // Validate arguments
+    if (startVersion < 0 || endVersion < startVersion) {
+      throw invalidVersionRange(startVersion, endVersion);
+    }
+
+    // Get any available commit files within the version range
+    List<FileStatus> commitFiles = listCommitFiles(engine, tablePath, startVersion, endVersion);
+
+    // There are no available commit files within the version range.
+    // This can be due to (1) an empty directory, (2) no valid delta files in the directory,
+    // (3) only delta files less than startVersion prefix (4) only delta files after endVersion
+    if (commitFiles.isEmpty()) {
+      throw noCommitFilesFoundForVersionRange(tablePath.toString(), startVersion, endVersion);
+    }
+
+    // Verify commit files found
+    // (check that they are continuous and start with startVersion and end with endVersion)
```

Comment:
btw, why such a strict requirement that they end with endVersion?

e.g. I ask for version in range [10, 1000] and you give me back versions from [10, 100] -> why should we error out?

do we expect users of this API to first check the latest end version available?

Reply:
I suppose it could be bad (and hard to tell that it is an error) if we return [10, 100] despite versions [101, 1000] existing.

so, this makes sense to me 👍

Reply:
> do we expect users of this API to first check the latest end version available?

Yes. Plus what you've said.